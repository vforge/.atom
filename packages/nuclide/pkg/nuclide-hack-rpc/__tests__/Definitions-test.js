"use strict";

function _Definitions() {
  const data = require("../lib/Definitions");

  _Definitions = function () {
    return data;
  };

  return data;
}

/**
 * Copyright (c) 2015-present, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the license found in the LICENSE file in
 * the root directory of this source tree.
 *
 * 
 * @format
 * @emails oncall+nuclide
 */
describe('convertDefinitions', () => {
  const filePath = '/tmp/file1';
  const defPath = '/tmp/file2';
  const projectRoot = '/tmp';
  const filteredDef = {
    definition_pos: null,
    name: 'filtered',
    pos: {
      filename: filePath,
      line: 1,
      char_start: 10,
      char_end: 13
    }
  };
  const def = {
    definition_pos: {
      filename: defPath,
      line: 10,
      char_start: 42,
      char_end: 52
    },
    definition_span: {
      filename: defPath,
      line_start: 10,
      char_start: 42,
      line_end: 11,
      char_end: 1
    },
    name: 'hack-name',
    pos: {
      filename: filePath,
      line: 1,
      char_start: 10,
      char_end: 13
    },
    projectRoot
  };
  const defaultName = {
    definition_pos: {
      filename: null,
      line: 10,
      char_start: 42,
      char_end: 52
    },
    name: 'hack-name',
    pos: {
      filename: filePath,
      line: 1,
      char_start: 10,
      char_end: 13
    },
    projectRoot
  };
  it('no definitions', () => {
    expect((0, _Definitions().convertDefinitions)([], filePath, projectRoot)).toBe(null);
  });
  it('filtered definitions', () => {
    expect((0, _Definitions().convertDefinitions)([filteredDef], filePath, projectRoot)).toBe(null);
  });
  it('valid definition', () => {
    expect((0, _Definitions().convertDefinitions)([filteredDef, def], filePath, projectRoot)).toMatchSnapshot();
  });
  it('default name definition', () => {
    expect((0, _Definitions().convertDefinitions)([defaultName], filePath, projectRoot)).toMatchSnapshot();
  });
});