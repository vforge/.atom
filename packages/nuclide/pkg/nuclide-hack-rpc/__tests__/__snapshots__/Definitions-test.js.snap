// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertDefinitions default name definition 1`] = `
Object {
  "definitions": Array [
    Object {
      "id": "hack-name",
      "language": "php",
      "name": "hack-name",
      "path": "/tmp/file1",
      "position": Point {
        "column": 41,
        "row": 9,
      },
      "projectRoot": "/tmp",
      "range": undefined,
    },
  ],
  "queryRange": Array [
    Range {
      "end": Point {
        "column": 13,
        "row": 0,
      },
      "start": Point {
        "column": 9,
        "row": 0,
      },
    },
  ],
}
`;

exports[`convertDefinitions valid definition 1`] = `
Object {
  "definitions": Array [
    Object {
      "id": "hack-name",
      "language": "php",
      "name": "hack-name",
      "path": "/tmp/file2",
      "position": Point {
        "column": 41,
        "row": 9,
      },
      "projectRoot": "/tmp",
      "range": Range {
        "end": Point {
          "column": 1,
          "row": 10,
        },
        "start": Point {
          "column": 41,
          "row": 9,
        },
      },
    },
  ],
  "queryRange": Array [
    Range {
      "end": Point {
        "column": 13,
        "row": 0,
      },
      "start": Point {
        "column": 9,
        "row": 0,
      },
    },
  ],
}
`;
